components/Componenten/tree-view/tree-view.njk

```diff
@@ -1,28 +1,28 @@
 {% macro leaves(items) %}
-  <ul>
-    {%- for item in items -%}
-      <li {{ className('dso-collection' if item.items else 'dso-entity', [item.active, 'dso-is-current']) }} >
-        {% if item.items %}
-          <button type="button">{{- item.label -}}</button>
-        {% else %}
-          <a href="#">{{- item.label -}}</a>
-        {% endif %}
-        {% if item.items %}
-          <ul>
-            {{ leaves(item.items) }}
-          </ul>
-        {% endif %}
-      </li>
-    {% endfor %}
-  </ul>
+  {%- for item in items -%}
+    <li {{ className('dso-collection' if item.items else 'dso-entity', [item.active, 'dso-is-current']) }} >
+      {% if item.items %}
+        <button type="button">{{- item.label -}}</button>
+      {% else %}
+        <a href="#">{{- item.label -}}</a>
+      {% endif %}
+      {% if item.items %}
+        <ul>
+          {{ leaves(item.items) }}
+        </ul>
+      {% endif %}
+    </li>
+  {% endfor %}
 {% endmacro %}
 
 <nav class="dso-tree-view">
   <ul>
     {%- for nav in navs %}
       <li>
         <h2 class="dso-tree-view-subhead">{{ nav.label }}</h2>
-        {{ leaves(nav.items) }}
+        <ul>
```

Comment:
Dit lost inderdaad het vorige probleem op, maar toch moet het volgens mij prima mogelijk zijn om `leaves()` altijd een `<ul>` te laten produceren (uiteraard check je dan eerst ff of er ueberhaupt `<li>` in gaan komen). Da's toch een stuk veiliger dan de verantwoordelijkheid voor de buitenste `<ul>` buiten de macro te leggen.

Reply:
Fixed